commands/cli.go

```diff
@@ -0,0 +1,290 @@
+						// if we have nothing at this phase, log an error and abort
+						if len(plaintext) == 0 {
+							log.Fatal("Empty input detected. Aborting")
+						}
+					}
+				}
+
+				secretName := c.Args().First()
+
+				secretEncoding := ""
+				// determine if we need to base64 the raw text, defaults
+				// to true.
+				if c.Bool("no-decode") == true {
+					// skip encoding, encode as plain value
+					secretEncoding = "plain"
+				} else {
+					// encode value as base64 compressed string
+					secretEncoding = "base64"
+					plaintext = []byte(base64.StdEncoding.EncodeToString(plaintext))
+				}
+
+				// Init a KMS client
+				client := cloud.NewGCPKMS(c.String("key"))
+
+				cypher, err := client.Encrypt(plaintext)
+				if err != nil {
+					log.Fatal(err)
+				}
+				// re-encode the binary data we got back.
+				encoded := base64.StdEncoding.EncodeToString(cypher)
+				toAdd := utils.Secret{
+					Name:       strings.ToUpper(secretName),
+					Cyphertext: encoded,
+					Created:    time.Now(),
+					Encoding:   secretEncoding,
+				}
+
+				utils.AddSecret(toAdd)
+
+				return nil
+			},
+		},
+		{
+			Name:  "send",
+			Usage: "Encode/Decode a secret for copy/paste",
+			Flags: []cli.Flag{
+				cli.StringFlag{
+					Name:   "key",
+					EnvVar: "SCTL_KEY",
+					Usage:  "GCloud KMS Key URI",
+				},
+			},
+			Action: func(c *cli.Context) error {
+
+				err := validateContext(c, "send")
+				if err != nil {
+					log.Fatalf("%s", err)
+				}
+				var plaintext []byte
+
+				// attempt stdin scan, SEND should be pipeable for things like cat'ing a file.
+				plaintext = stdinScan()
+				if plaintext == nil {
+					plaintext = utils.UserInput()
+				}
+				if len(plaintext) == 0 {
+					log.Fatal("Empty input detected. Aborting")
+				}
+
+				client := cloud.NewGCPKMS(c.String("key"))
+				cypher, err := client.Encrypt(plaintext)
+				if err != nil {
+					log.Fatal(err)
+				}
+				encoded := base64.StdEncoding.EncodeToString(cypher)
+
+				fmt.Println("Hello, I've shared some data with you with sctl! https://github.com/vapor-ware/sctl")
+				fmt.Println("Once installed, run the following commands to view this sensitive information")
+				fmt.Println("")
+				fmt.Println("```")
+				cmd := fmt.Sprintf("sctl receive --key=%s %s", c.String("key"), encoded)
+				fmt.Println(cmd)
+				fmt.Println("```")
+				return nil
+			},
+		},
+		{
+			Name:  "receive",
+			Usage: "Read a plaintext encoded secret",
+			Flags: []cli.Flag{
+				cli.StringFlag{
+					Name:   "key",
+					EnvVar: "SCTL_KEY",
+					Usage:  "Gcloud KMS Key URI",
+				},
+			},
+			Action: func(c *cli.Context) error {
+				if len(c.Args()) >= 1 {
+					decoded, err := base64.StdEncoding.DecodeString(c.Args().First())
+					if err != nil {
+						log.Fatal(err)
+					}
+					client := cloud.NewGCPKMS(c.String("key"))
+					cypher, err := client.Decrypt(decoded)
+					if err != nil {
+						log.Fatal(err)
+					}
+
+					fmt.Println(string(cypher))
+				}
```

Comment:
perhaps outside the scope of this PR, but shouldn't we raise an error if no args were provided?

also, it looks like this checks to see if there is >= 1 arg, but it only ever uses the first arg? if we never use more than 1 arg, it should probably also error if multiple args are given

Reply:
i think the userinput() validations are sufficient here?